logbook-core/src/main/java/org/zalando/logbook/core/Conditions.java

```diff
@@ -94,4 +96,14 @@ public static <T extends HttpMessage> Predicate<T> withoutHeader(final String ke
         return message -> !message.getHeaders().containsKey(key);
     }
 
+    public static <T extends HttpRequest> Predicate<T> conditionalHeader(
```

Comment:
Can't say I'm convinced that this separate method is needed to be honest. You can achieve the same result with `header(final String key, final Predicate<String> predicate)` method.

Reply:
`conditionalHeader` method wraps `withoutHeader` and `header` logic, or it can be extracted outside, whatever you prefer just let me know.